### PR TITLE
Port download helper from CI3

### PR DESCRIPTION
**Pull request by jim-parry**
Needed for Compression module.
CI3 had skipped unit tests, reproduced here... will need fleshing out at some point.

### CODE REVIEW

**Comment by jim-parry** [comment]
The unit tests now show a number of risky tests - output buffers not closed!?
Another PR has caused these - I have made no changes to those tests :(

File: system/Helpers/download_helper.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	 * Generates headers that force a download to happen
+	 *
+	 * @param	mixed	filename (or an array of local file path => destination filename)
+	 * @param	mixed	the data to be downloaded
+	 * @param	bool	whether to try and send the actual file MIME type
+	 * @return	void
+	 */
+	function force_download($filename = '', $data = '', $set_mime = FALSE)
+	{
+		if ($filename === '' OR $data === '')
```
Comment (ytetsuro) [inline system/Helpers/download_helper.php]: Is it impossible to realize with the following code?

```php
CodeIgniter\Config\Services::response()->download($filename, $data, $set_mime);
```

https://github.com/bcit-ci/CodeIgniter4/blob/develop/system/HTTP/Response.php#L943

Comment (jim-parry) [reply system/Helpers/download_helper.php]: It may very well be - I will check :) Thank you!

**Comment by jim-parry** [comment]
@ytetsuro That does solve my problem, thanks.

**Pull request closed without merging**